InternalPythonModules/android/imo.py

```diff
@@ -0,0 +1,122 @@
+limitations under the License.
+"""
+
+from java.io import File
+from java.lang import Class
+from java.lang import ClassNotFoundException
+from java.lang import Long
+from java.lang import String
+from java.sql import ResultSet
+from java.sql import SQLException
+from java.sql import Statement
+from java.util.logging import Level
+from java.util import ArrayList
+from org.apache.commons.codec.binary import Base64
+from org.sleuthkit.autopsy.casemodule import Case
+from org.sleuthkit.autopsy.coreutils import Logger
+from org.sleuthkit.autopsy.coreutils import MessageNotifyUtil
+from org.sleuthkit.autopsy.coreutils import AppSQLiteDB
+from org.sleuthkit.autopsy.coreutils import AppDBParserHelper
+from org.sleuthkit.autopsy.datamodel import ContentUtils
+from org.sleuthkit.autopsy.ingest import IngestJobContext
+from org.sleuthkit.datamodel import AbstractFile
+from org.sleuthkit.datamodel import BlackboardArtifact
+from org.sleuthkit.datamodel import BlackboardAttribute
+from org.sleuthkit.datamodel import Content
+from org.sleuthkit.datamodel import TskCoreException
+from org.sleuthkit.datamodel import Account
+
+import traceback
+import general
+
+"""
+Finds the SQLite DB for IMO, parses the DB for contacts & messages,
+and adds artifacts to the case.
+"""
+class IMOAnalyzer(general.AndroidComponentAnalyzer):
+    def __init__(self):
+        self._logger = Logger.getLogger(self.__class__.__name__)
+
+    def analyze(self, dataSource, fileManager, context):
+        try:
+            selfAccountId = None
+            accountDbs = AppSQLiteDB.findAppDatabases(dataSource, "accountdb.db", "com.imo.android.imous")
+            for accountDb in accountDbs:
+                accountResultSet = accountDb.runQuery("SELECT uid, name FROM account")
+                if accountResultSet:
+                    # We can determine the IMO user ID of the device owner. 
+                    # Therefore we can create and use a app account and use that 
+                    # as a 'self' account instead of a Device account
+                    if not selfAccountId:
+                        selfAccountId = accountResultSet.getString("name")
+                    accountResultSet.close()
+                accountDb.close();
+                        
+            friendsDbs = AppSQLiteDB.findAppDatabases(dataSource, "imofriends.db", "com.imo.android.imous")
+            for friendsDb in friendsDbs:
+                friendsDBHelper = AppDBParserHelper("IMO Parser", friendsDb.getDBFile(),
+                                                    Account.Type.MESSAGING_APP, Account.Type.MESSAGING_APP, selfAccountId )
+                contactsResultSet = friendsDb.runQuery("SELECT buid, name FROM friends")
+                if contactsResultSet is not None:
+                    while contactsResultSet.next():
+                        friendsDBHelper.addContact( contactsResultSet.getString("name"),  ## name for account
```

Comment:
How is the name unique?

Reply:
It's not.  That was one of the open questions we discussed today.  Although then we decided to create the account with the id, and attach the name as an attribute to it, since we don't want  to expose all the account creation/modification to the parsing module, we are going to defer that.

The name is more useful than the buid.